webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala

```diff
@@ -183,6 +183,16 @@ final case class NoReportInInterval(
     expirationDateTime: DateTime
 ) extends NoReport
 
+/*
+ * When we want to keep compliance even in case of NoReportInterval
+ */
+final case class KeepLastCompliance(
+    expectedConfig:     NodeExpectedReports,
+    expiredSince:       DateTime,
+    expirationDateTime: DateTime,
```

Comment:
I'm struggling to understand the difference between the two fields only from the naming, what is `expirationDateTime` if it's not the date since which compliance expire ?

Reply:
is it not `expirationTime: DateTime, keepUntil: DateTime` ?

Reply:
or `expiredSince: DateTime, keepUntil: DateTime` ?

Reply:
it's what I did initially, but `expirationDateTime` has a special meaning provided by the trait `ExpiringStatus` and is used by the compute engine to know when the node need to be checked again if nothing happened to it. 
So in the case of `KeepLastCompliance`, it need to be the time on which the compliance that we kept should not be kept anymore. 

But I wanted to also have at hand the time on which the node should have become grey to make informative message, like "that node stopped responding at $expiredSince, but that's ok because it is configured to have extra grace period until $expirationDateTime"

Reply:
ok ! it's makes sense for me now !